cardano-cli/test/cardano-cli-test/Test/Cli/Parser.hs

```diff
@@ -97,3 +108,31 @@ hprop_integral_pair_reader_negative = propertyOnce $ do
     case Parsec.runParser pairIntegralParsecParser () "" s of
       Left parsecError -> Left $ show parsecError
       Right x -> Right x
+
+-- | Execute me with:
+-- @cabal test cardano-cli-test --test-options '-p "/cddl tx envelope completeness/"'@
+hprop_test_cddl_tx_envelope_completeness :: Property
+hprop_test_cddl_tx_envelope_completeness = property $ do
+  AnyShelleyBasedEra era <- H.noteShowM . forAll $ Gen.element [minBound .. maxBound]
+  x <- forAll $ genTx era
+  shelleyBasedEraConstraints era $ do
+    let TextEnvelopeType d = textEnvelopeType (proxyToAsType (getProxy x))
```

Comment:
I'm not sure what this test is testing. Expected `txTextEnvelopeTypes` is generated using `textEnvelopeType` so this will always succeed.

Reply:
It ensures that the list has all the `TextEnvelopeTypes`, see this comment: https://github.com/IntersectMBO/cardano-cli/pull/892#discussion_r1761414468.

Now, the implementation gets updated automatically and it is pretty similar to the test so it doesn't make that much sense anymore.

Initially it wasn't, the list was a literal, so I didn't know to update it and that caused a bug.

Also the list could contain more things (like it did before), the test checks it at least contains the `TextEnvelopeTypes`.

But we can remove the tests if they don't make sense anymore

Reply:
I'm for removing them. The expected values `txTextEnvelopeTypes` are generated using `HasTextEnvelope`. The envelopes for `Tx` are also coming from `HasTextEnvelope`. Does not make much sense to me.

Reply:
Removed them here: https://github.com/IntersectMBO/cardano-cli/pull/892/commits/85b74885cb02e907acec7ed36d91200d79121ca1